### PR TITLE
21929 - get reviews endpoint

### PR DESCRIPTION
**Pull request by ketaki-deodhar**
*Issue #:* /bcgov/entity#21929

*Description of changes:*

- "get reviews" endpoint
- support pagination (added logic to handle limit and page)
- add unit tests

image

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the lear license (Apache 2.0).

### CODE REVIEW

**Comment by ketaki-deodhar** [comment]
WIP for unit tests. added 2 and they run successfully

File: legal-api/src/legal_api/resources/v2/admin/reviews.py [hunk 1]
```diff
@@ -14,15 +14,42 @@
...
 
 
+@bp_admin.route('/reviews', methods=['GET'])
+@cross_origin(origin='*')
+@jwt.has_one_of_roles([UserRoles.staff])
+def get_reviews():
+    """Return a list of reviews."""
+    page = int(request.args.get('page', 1))
+    limit = int(request.args.get('limit', 10))
+    query = db.session.query(Review).order_by(Review.submission_date.asc())
```
Comment (vysakh-menon-aot) [inline legal-api/src/legal_api/resources/v2/admin/reviews.py:33]: use creation_date for sorting as default

File: legal-api/src/legal_api/resources/v2/admin/reviews.py [hunk 2]
```diff
@@ -14,15 +14,42 @@
...
+    reviews_list = [
+        review.json for review in results
+    ]
+
+    reviews = {
+        'reviews': reviews_list,
+        'page': page,
+        'limit': limit,
+        'total': total_count
+    }
```
Comment (vysakh-menon-aot) [inline legal-api/src/legal_api/resources/v2/admin/reviews.py:48]: can you move this into the review model as a new function

File: legal-api/tests/unit/resources/v2/test_reviews.py [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    assert rv.status_code == HTTPStatus.UNAUTHORIZED
+
+def test_get_reviews_with_valid_user(app, session, client, jwt):
+    """Assert review object returned for STAFF role."""
+
+    rv = client.get(f'/api/v2/admin/reviews',
+                    headers=create_header(jwt, [STAFF_ROLE], 'user'))
+
+    assert rv.status_code == HTTPStatus.OK
+    assert 'reviews' in rv.json
```
Comment (vysakh-menon-aot) [inline legal-api/tests/unit/resources/v2/test_reviews.py:41]: can you add some data to review table to verify this?

File: legal-api/src/legal_api/models/review.py [hunk 4]
```diff
@@ -76,6 +76,27 @@ def get_review(cls, filing_id) -> Review:
                       one_or_none())
         return review
 
+    @classmethod
+    def get_paginated_reviews(cls, page, limit):
+        """Return paginated reviews."""
+        query = db.session.query(Review).order_by(Review.creation_date.asc())
```
Comment (severinbeauvais) [inline legal-api/src/legal_api/models/review.py:82]: 👍 Yes, the default sort is oldest first. This is probably worth a comment, but then again, this code will be expanded soon to handle filtering and sorting by different fields.

**Review by severinbeauvais — COMMENTED** [review]
LGTM. I agree with Vysakh that it would be nice to have some test data in here.

Vysakh, could we merge this now (which would allow Arwen to call this) and then work on the test data?

**Review by severinbeauvais — APPROVED** [review]

File: legal-api/src/legal_api/resources/v2/admin/reviews.py [hunk 5]
```diff
@@ -23,6 +23,18 @@
...
+@jwt.has_one_of_roles([UserRoles.staff])
+def get_reviews():
+    """Return a list of reviews."""
+    page = int(request.args.get('page', 1))
+    limit = int(request.args.get('limit', 10))
+    reviews = Review.get_paginated_reviews(page, limit)
+
+    return reviews, HTTPStatus.OK
+
+
```
[the remark this answers is not in the document]
Comment (ketaki-deodhar) [reply legal-api/src/legal_api/resources/v2/admin/reviews.py:37]: Yes. Only callable by staff

File: legal-api/tests/unit/resources/v2/test_reviews.py (hunk 3 above)
Comment (ketaki-deodhar) [reply legal-api/tests/unit/resources/v2/test_reviews.py:41]: Working on it

Comment (severinbeauvais) [reply legal-api/tests/unit/resources/v2/test_reviews.py:41]: Do you have an initial view of what the future arguments should look like?
- sorting by different fields
- filtering on different fields

**Review by ArwenQin — APPROVED** [review]
Thanks, LGTM

**Review by vysakh-menon-aot — APPROVED** [review]

**Pull request merged**